client/src/components/leftPanel/LeftPanel.js

```diff
@@ -5,18 +5,32 @@ import './LeftPanel.scss'
 
 class LeftPanel extends Component {
 
+  state = {
+    searchString: ""
+  }
+
+  handleChange = (event) => this.setState({ searchString: event.target.value })
+  
   render() {
     const {tags} = this.props
+    const searchString = this.state.searchString.trim().toLowerCase();
```

Comment:
I think it hurts the readability, that you use the name `searchString` for different things. I would be fine with something like `lowerCaseSearchString` here.

..and if you do this, you can go `const {searchString} = this.state` and use the _real_ `searchString` in the places where it is needed.